### DISCUSSION TITLE
Huawei Watch 3 Pro

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#2169] [Q&A]
Hey,
Has anyone experience with the Sideload of xDrip+ on the Huawei Watch 3?
I want to buy a watch, and I like this watch...
But I need xDrip on it!

**Comment by Miquest** [comment]
There is a Guide for App Sideload with ADB, but I read something about problems with sideloading...
xDrip isn't functioning in some Watches

**Reply by Navid200** [reply]
Have you looked here:  https://xdrip.readthedocs.io/en/latest/smartwatch/smartwatch/

**Reply by Miquest** [reply]
Yes, but the Huawei Watch 3 has HarmonyOS as operating system, and not Android Wear...
That's why I asked for expirences from people who have a Huawei Watch 3....
It's very important for me that I can see my blood sugar on my smartwatch.
And I can't use Wear OS....

**Comment by yuri-vashchenko** [comment]
I have Huawei Watch 3 Pro and would like to have it supported.
If any help like being a beta tester is needed I'm ready.

**Reply by Miquest** [reply]
Have you tried the APK Sideload with ADB?
And does it work?

**Reply by yuri-vashchenko** [reply]
@thorinEichenschild007 , does APK Sideload work on Harmony OS at all?

**Reply by Miquest** [reply]
https://www.huaweiblog.de/news/anleitung-huawei-watch-3-apps-apk-installieren/

Here are the instructions I read...
HarmonyOS is like Android Wear, with the ADB Debugging via Wifi...

**Reply by yuri-vashchenko** [reply]
I was able to install xDrip+ on my watch via ADB, but it does not work - just crashes after I start it and accept license

**Reply by Miquest** [reply]
Which Sensor are you using?
And does it crash totally or just don't receive readings from the sensor?

**Reply by yuri-vashchenko** [reply]
I'm using Libre 1 with Bubble Mini.
The xDrip+ application just silently closes without any error messages.
I was not even able to start setting it up. All I could do before it started crashing all the time was accepting the license.

**Reply by Miquest** [reply]
@jamorham
Is there a SDK for HarmonyOS on the Huawei Watches?

**Reply by floosde** [reply]
@thorinEichenschild007 , yes, DevEco IDE
I want to support topic. A properly working app or at least a watchface is needed for my Huawei Watch 3 Pro

**Comment by floosde** [comment]
Does anyone have at least an idea how to make normal alerts on Huawei Watch 3 (Pro). Because just one small vibration isn't that good for alerts as sounds on mobile phone.

**Reply by Miquest** [reply]
You were able to set up xDrip Plus on Huawei Watch 3?
How did you do this?

**Comment by yuri-vashchenko** [comment]
On my Huawei Watch 3 I was able to:

1. Install Firefox browser on my watch and open nightscout web on watch - I can see my blood sugar when I open the browser, but it does not run in background nor generates any alerts/events.
2. Install Nsclient app o watch - It has similar to AndroidAPS application, it shows blood sugar too and can generate alerts, but it does not work in background - to have actual values you need to reopen app often and keep it in foreground
3. I was **not** able to install / run any system service or watchface showing my blood sugar in realtime or wake me up at night if my blood sugar goes out of limits.

**Reply by Miquest** [reply]
Nice!
Thank you for the detailed guide!
But with this method, the watch isn't able to collect data as a standalone collector, is it?

**Comment by MarcinLachowicz** [comment]
Hello, I've just installed xDrip on my phone and I would like to have it also on my Huawei Watch 3. Does anyone investigated this topic a little bit more?
I tried tried to install WearOS app on my Huawei, but it wasnt possible due to lack of some libraries in HarmonyOS
